WIP Run BYOR install on installer presubmits

vrutkovs opened the pull request:
This PR would ensure openshift-ansible is not broken by installer changes.
`always_run` is set to False, so it would be started by request for now.

PTAL @wking @crawford

wking commented on ci-operator/config/openshift/installer/openshift-installer-master.yaml:
```diff
@@ -78,3 +78,7 @@ tests:
+- as: e2e-byor
+  commands: TEST_SUITE=openshift/smoke-4 run-tests run-tests
+  openshift_ansible_40:
+    cluster_profile: gcp-40
```
Looks like you're missing a trailing newline here.

wking commented on ci-operator/config/openshift/installer/openshift-installer-master.yaml:
```diff
@@ -78,3 +78,7 @@ tests:
   openshift_installer:
     cluster_profile: openstack
+- as: e2e-byor
+  commands: TEST_SUITE=openshift/smoke-4 run-tests run-tests
```
Looks like these `run-tests` doublings were typos, see #2278.

wking commented on ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml:
```diff
@@ -115,6 +115,66 @@ presubmits:
+        - name: RPM_REPO_OPENSHIFT_ORIGIN
+          value: https://rpms.svc.ci.openshift.org/openshift-origin-v4.0/
+        - name: TEST_COMMAND
+          value: TEST_SUITE=openshift/smoke-4 run-tests run-tests
```
Here's another `run-tests` doubling.

stevekuznetsov commented:
Why not test these using the local testing tools before merge? :)

vrutkovs commented:
I'll mark this is as WIP as it would currently always fail due to changes in https://github.com/openshift/installer/pull/955

vrutkovs commented:
Fixed syntax error, duplicated `run-tests` and added `optional`.

Two more PRs are required to be merged for this test to pass:
* [ ] https://github.com/openshift/machine-config-operator/pull/266
* [ ] https://github.com/openshift/release/pull/2504

vrutkovs commented:
This is no longer planned
